tests/test_app.py

```diff
@@ -0,0 +1,680 @@
+        'password': 'testpassword'
+    }
+    response = suite['testapp'].post('/login', params=params, status='3*')
+    redirected = response.follow()
+
+    assert LOGOUT in redirected.body and '<main id="home">' in redirected.body
+    assert LOGIN not in redirected.body
+
+
+# Test 7
+# use wrong password
+def test_login_bad_pass_unittest(suite):
+    params = {
+        'username': 'Test_Username',
+        'password': 'oops'
+    }
+    suite['auth_req'].params = params
+    assert not app.login(suite['auth_req'])
+
+
+# Test 8
+# use wrong password
+def test_login_bad_pass(suite):
+    params = {
+        'username': 'Test_Username',
+        'password': 'oops'
+    }
+    response = suite['testapp'].post('/login', params=params, status='2*')
+
+    assert 'Not authenticated' in response.body
+
+
+# Test 9
+# use wrong username
+def test_login_bad_user_unittest(suite):
+    params = {
+        'username': 'whoops',
+        'password': 'testpassword'
+    }
+    suite['auth_req'].params = params
+    with pytest.raises(ValueError):    # login() raises ValueError
+        app.login(suite['auth_req'])
+
+
+# Test 10
+# use wrong username
+def test_login_bad_user(suite):
+    params = {
+        'username': 'whoops',
+        'password': 'testpassword'
+    }
+    response = suite['testapp'].post('/login', params=params, status='2*')
+
+    assert 'User does not exist' in response.body
+
+
+# Test 11
+# try logging in without all params
+def test_login_missing_params_unittest(suite):
+    params = [
+        {'username': 'Test_Username'},
+        {'password': 'testpassword'}
+    ]
+    for param in params:
+        suite['auth_req'].params = param
+        with pytest.raises(ValueError):
+            app.login(suite['auth_req'])
+
+
+# Test 12
+# try logging in without password
+def test_login_missing_password(suite):
+    params = {
+        'username': 'Test_Username'
+    }
+    response = suite['testapp'].post('/login', params=params, status='2*')
+
+    assert 'Username and password are required' in response.body
+
+
+# Test 13
+# try logging in without username
+def test_login_missing_username(suite):
+    params = {
+        'password': 'testpassword'
+    }
+    response = suite['testapp'].post('/login', params=params, status='2*')
+
+    assert 'Username and password are required' in response.body
+
+
+# Test 14
+# logout logged in user
+def test_logout(suite):
+    test_login_success(suite)
+    redirect = suite['testapp'].get('/logout', status="3*")
+    response = redirect.follow()
+    assert LOGIN in response.body and '<main id="home">' in response.body
+    assert LOGOUT not in response.body
+
+
+# Test 15
+# try to create user without completing captcha
+def test_create_user_without_captcha(suite):
+    params = {
+        'username': 'Test_Username',
+        'password': 'testpassword',
+        'confirm': 'testpassword'
+    }
+    response = suite['testapp'].post('/new_account', params=params, status='2*')
```

Comment:
Does this test actually hit the google recaptcha servers?  That's the type of I/O dependency that `mock` would help you to avoid.